topics/functions-intro.md

```diff
@@ -24,68 +27,55 @@ This approach suits many light-weight scripting use cases, but introduces severa
 
 Functions also simplify development by enabling code sharing.
-Every function belongs to a single library, and any given library can consist of multiple functions.
+Every function has a user-defined name and belongs to a library, and a library can consist of multiple functions.
 The library's contents are immutable, and selective updates of its functions aren't allowed.
 Instead, libraries are updated as a whole with all of their functions together in one operation.
 This allows calling functions from other functions within the same library, or sharing code between functions by using a common code in library-internal methods, that can also take language native arguments.
 
-Functions are intended to better support the use case of maintaining a consistent view for data entities through a logical schema, as mentioned above.
-As such, functions are stored alongside the data itself.
-Functions are also persisted to the AOF file and replicated from primary to replicas, so they are as durable as the data itself.
-When Valkey is used as an ephemeral cache, additional mechanisms (described below) are required to make functions more durable.
-
 Like all other operations in Valkey, the execution of a function is atomic.
 A function's execution blocks all server activities during its entire time, similarly to the semantics of [transactions](transactions.md).
 These semantics mean that all of the script's effects either have yet to happen or had already happened.
 The blocking semantics of an executed function apply to all connected clients at all times.
 Because running a function blocks the Valkey server, functions are meant to finish executing quickly, so you should avoid using long-running functions.
 
+Functions are written in [Lua 5.1](lua-api.md).
+Valkey functions can use all of Lua's available capabilities to ephemeral scripts,
+with the only exception being the [Valkey Lua scripts debugger](ldb.md).
+In the future, it may be possible to extend Valkey to allow other languages.
+
 ## Loading libraries and functions
 
 Let's explore Valkey Functions via some tangible examples and Lua snippets.
 
 At this point, if you're unfamiliar with Lua in general and specifically in Valkey, you may benefit from reviewing some of the examples in [Introduction to Eval Scripts](eval-intro.md) and [Lua API](lua-api.md) pages for a better grasp of the language.
 
 Every Valkey function belongs to a single library that's loaded to Valkey.
-Loading a library to the database is done with the `FUNCTION LOAD` command.
-The command gets the library payload as input,
-the library payload must start with Shebang statement that provides a metadata about the library (like the engine to use and the library name).
+Loading a library to the database is done with the [`FUNCTION LOAD`](../commands/function-load.md) command.
+The command takes the library source code as input and
```

Comment:
TBH the sentence is a bit broken. I think "the" is redundant here. What is meant under "It"? The library, the command?

Reply:
Which "the" is redundant? The second one, "the library source code"?

There is no "It" in the text you marked, but I assume you mean "it must start with a Shebang". That's the source code. Can a command start with a shebang? The grammar is maybe ambiguous but isn't it clear from the context?

Reply:
Actually, I want to delete "The command takes the library source code as input" because it doesn't say anything. The previous sentence already says that loading a library is done with this command.

Updated suggestion:

> Every Valkey function belongs to a single library that's loaded to Valkey.
Loading a library to the database is done with the [`FUNCTION LOAD`](../commands/function-load.md) command.
The library source code must start with a Shebang line that provides metadata about the library, like the language (always "lua") and the library name.
The Shebang format is:

Reply:
As a suggestion, will it make sense to split the sentence in two? E.g. "The command takes the library source code as input. The source code must start with a Shebang line that provides the metadata about the library, like the language (always "lua") and the library name." 

This will help inexperienced users understand what this is all about:)

Reply:
> Actually, I want to delete "The command takes the library source code as input" because it doesn't say anything. The previous sentence already says that loading a library is done with this command.
> 
> Updated suggestion:
> 
> > Every Valkey function belongs to a library that's loaded to Valkey.
> > Loading a library to the database is done with the [`FUNCTION LOAD`](../commands/function-load.md) command.
> > The library source code must start with a Shebang line that provides metadata about the library, like the language (always "lua") and the library name.
> > The Shebang format is:

Yes, I like this variant.